### PR TITLE
Add Support for Ingress Egress policies

### PR DESCRIPTION
**Pull request by lcaggio**

### CODE REVIEW

File: modules/vpc-sc/main.tf [hunk 1]
```diff
@@ -89,6 +93,93 @@ resource "google_access_context_manager_service_perimeter" "standard" {
...
+
+                content {
+                  service_name = try(operations.value.service_name, null)
+
+                  dynamic "method_selectors" {
+                    for_each = try([operations.value.method_selectors], [])
+
+                    content {
+                      method     = try(method_selectors.value.method, null)
+                      permission = try(method_selectors.value.permission, null)
```
Comment (drebes) [inline modules/vpc-sc/main.tf:126]: I was surprised by this. I see it supported on the provider resource, but it is not listed on the docs. Have you tested this?

File: modules/vpc-sc/main.tf [hunk 2]
```diff
@@ -89,6 +93,93 @@ resource "google_access_context_manager_service_perimeter" "standard" {
...
+        content {
+          dynamic "egress_from" {
+            for_each = try(var.egress_policies[egress_policies.value].egress_from != null ? [""] : [], [])
+
+            content {
+              identity_type = try(var.egress_policies[egress_policies.value].egress_from.identity_type, null)
+              identities    = try(var.egress_policies[egress_policies.value].egress_from.identities, null)
+            }
+          }
+          dynamic "egress_to" {
```
Comment (drebes) [inline modules/vpc-sc/main.tf:109]: Likewise, maybe skip the dynamic block.

File: modules/vpc-sc/main.tf [hunk 3]
```diff
@@ -89,6 +93,93 @@ resource "google_access_context_manager_service_perimeter" "standard" {
...
+              resources = try(var.egress_policies[egress_policies.value].egress_to.resources, null)
+
+              dynamic "operations" {
+                for_each = try(var.egress_policies[egress_policies.value].egress_to.operations, [])
+
+                content {
+                  service_name = try(operations.value.service_name, null)
+
+                  dynamic "method_selectors" {
+                    for_each = try([operations.value.method_selectors], [])
```
Comment (drebes) [inline modules/vpc-sc/main.tf:122]: Have you tried an example with two `method_selectors`? For me this only works if I remove the [] from here and add it to the variable definition.

File: modules/vpc-sc/main.tf (hunk 1 above)
Comment (drebes) [reply modules/vpc-sc/main.tf:126]: Found it on the docs, cool!

File: modules/vpc-sc/main.tf [hunk 4]
```diff
@@ -89,6 +93,93 @@ resource "google_access_context_manager_service_perimeter" "standard" {
           allowed_services   = each.value.enforced_config.vpc_accessible_services
         }
       }
+
+      dynamic "egress_policies" {
+        for_each = try(local.perimeter_egress_policies_enforced[each.key] != null ? local.perimeter_egress_policies_enforced[each.key] : [], [])
+
+        content {
+          dynamic "egress_from" {
```
[the remark this answers is not in the document]
Comment (lcaggio) [reply modules/vpc-sc/main.tf:101]: Those are not required over API, I did a test and it works. I'll keep it, we can change the behaviour in a future iteration.

File: modules/vpc-sc/main.tf (hunk 2 above)
Comment (lcaggio) [reply modules/vpc-sc/main.tf:109]: Those are not required over API, I did a test and it works. I'll keep it, we can change the behaviour in a future iteration

File: modules/vpc-sc/main.tf (hunk 3 above)
Comment (lcaggio) [reply modules/vpc-sc/main.tf]: you are right, fixed

**Review by drebes — APPROVED** [review]
LGTM

**Pull request merged**